### PR TITLE
image-builder: improve sentry integration

### PR DESCRIPTION
**Pull request by diaasami**
Add a logrus hook to send Error logs to sentry as well
https://issues.redhat.com/browse/HMS-1986

### CODE REVIEW

File: cmd/image-builder/main.go [hunk 1]
```diff
@@ -46,6 +47,7 @@ func main() {
 		if err != nil {
 			panic(err)
 		}
+		defer sentry.Flush(5 * time.Second)
```
Comment (croissanne) [inline cmd/image-builder/main.go:51]: I'm not sure I understand the purpose of this, if it panics will it still be executed even with the 5 seconds timer?

File: cmd/image-builder/main.go [hunk 2]
```diff
@@ -57,6 +58,12 @@ func main() {
 
 	if conf.GlitchTipDSN == "" {
 		logrus.Warn("Sentry/Glitchtip was not initialized")
+	} else {
+		sentryhook := sentrylogrus.NewFromClient([]logrus.Level{logrus.PanicLevel,
+			logrus.FatalLevel, logrus.ErrorLevel},
+			sentry.CurrentHub().Client())
+		logrus.AddHook(sentryhook)
+		defer sentryhook.Flush(2 * time.Second)
```
Comment (croissanne) [inline cmd/image-builder/main.go:66]: Why is the flush needed here?

File: cmd/image-builder/main.go [hunk 3]
```diff
@@ -60,7 +60,7 @@ func main() {
 		logrus.Warn("Sentry/Glitchtip was not initialized")
 	} else {
 		sentryhook := sentrylogrus.NewFromClient([]logrus.Level{logrus.PanicLevel,
-			logrus.FatalLevel, logrus.ErrorLevel},
+			logrus.FatalLevel, logrus.ErrorLevel, logrus.InfoLevel},
```
Comment (croissanne) [inline cmd/image-builder/main.go:63]: This seems very overkill, the point of glitchtip is to only get the few errors that are so rare they might not influence the SLOs.

File: cmd/image-builder/main.go (hunk 1 above)
Comment (diaasami) [reply cmd/image-builder/main.go:51]: it takes a maximum 5 seconds to flush existing channels before main ends, but might return before the 5 seconds.

File: cmd/image-builder/main.go (hunk 3 above)
Comment (diaasami) [reply cmd/image-builder/main.go:63]: the `InfoLevel` part will be reverted later, that's why it's a separate commit, only meant for testing to help us make sure everything is working.

File: cmd/image-builder/main.go (hunk 2 above)
Comment (diaasami) [reply cmd/image-builder/main.go:66]: to process logrushook channels (going into sentry's channels), I believe.

**Review by croissanne — APPROVED** [review]

**Pull request merged**